src/Nethermind/Nethermind.BeaconNode/MockedStart/QuickStartServiceCollectionExtensions.cs

```diff
@@ -43,14 +45,32 @@ public static void AddQuickStart(this IServiceCollection services, IConfiguratio
                     x.Eth1BlockHash = new Hash32(section.GetBytesFromPrefixedHex("Eth1BlockHash", () => s_defaultEth1BlockHash));
                     x.Eth1Timestamp = section.GetValue("Eth1Timestamp", DefaultEth1Timestamp);
                     x.UseSystemClock = section.GetValue<bool>("UseSystemClock");
+                    x.ValidatorStartIndex = section.GetValue<ulong>("ValidatorStartIndex");
+                    x.NumberOfValidators = section.GetValue<ulong>("NumberOfValidators");
+                    x.ClockOffset = section.GetValue<long>("ClockOffset");
                 });
             });
 
             if (!configuration.GetValue<bool>("QuickStart:UseSystemClock"))
             {
-                ulong genesisTime = configuration.GetValue<ulong>("QuickStart:GenesisTime");
-                QuickStartClock quickStartClock = new QuickStartClock(genesisTime);
-                services.AddSingleton<IClock>(quickStartClock);
+                long clockOffset;
+                if (configuration.GetSection("QuickStart:ClockOffset").Exists())
+                {
+                    clockOffset = configuration.GetValue<long>("QuickStart:ClockOffset");
+                }
+                else
+                {
+                    ulong genesisTime = configuration.GetValue<ulong>("QuickStart:GenesisTime");
+                    clockOffset = (long) genesisTime - DateTimeOffset.Now.ToUnixTimeSeconds();
```

Comment:
you are using the Clock class elsewhere but not here, is that fine?

Reply:
This is for configuring / creating an alternative clock (QuickStartClock) that runs in real time but starts from a different offset. Used for running a specific fake date. Originally it was so that you could pass in a genesis time to beacon node (as per the interop quick start) and the clock would start from that point in time.  

It is only for testing, and may not end up very useful (in which case we can just remove it).  I'm not really sure how other implementations handle the quick start interop.

But to answer your question it is fine here, as it is creating an IClock (so can't use it yet).